### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by Mandybur800**

### CODE REVIEW

File: src/main/java/ua/com/conductor/dao/MovieDao.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package ua.com.conductor.dao;
+
+import java.util.List;
+import ua.com.conductor.model.Movie;
+
+public interface MovieDao {
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/dao/MovieDao.java:6]: suggestion: in future we will have multiple Dao, as well as multiple Services. I think it's better to start using Generic Interface principle

File: src/main/java/ua/com/conductor/dao/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import ua.com.conductor.lib.Dao;
+import ua.com.conductor.model.Movie;
+import ua.com.conductor.util.HibernateUtil;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/dao/MovieDaoImpl.java:15]: look [closely](https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html)

File: src/main/java/ua/com/conductor/dao/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (Max-Pochepets) [inline src/main/java/ua/com/conductor/dao/MovieDaoImpl.java]: as well as here

File: src/main/resources/hibernate.cfg.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        <property name = "hibernate.connection.url">jdbc:mysql://localhost/cinema?serverTimezone=UTC</property>
+        <property name = "hibernate.connection.username">root</property>
+        <property name = "hibernate.connection.password">1234</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+
+        <mapping class="ua.com.conductor.model.Movie"/>
+
+    </session-factory>
+</hibernate-configuration>
```
Comment (Max-Pochepets) [inline src/main/resources/hibernate.cfg.xml]: is there an empty line here?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]
👍 few minor steps

File: src/main/java/ua/com/conductor/dao/MovieDao.java (hunk 1 above)
Comment (Mandybur800) [reply src/main/java/ua/com/conductor/dao/MovieDao.java:6]: For this part of hw we don't need generic dao or service. Bohdan says it today.

File: src/main/resources/hibernate.cfg.xml (hunk 4 above)
Comment (Mandybur800) [reply src/main/resources/hibernate.cfg.xml:18]: as for me, if I have blank space it will be more readable

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/ua/com/conductor/dao/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert movie entity", e);
```
Comment (Sofasmile) [inline src/main/java/ua/com/conductor/dao/MovieDaoImpl.java]: let's create own exception here)

File: src/main/java/ua/com/conductor/dao/MovieDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie entity");
```
Comment (Sofasmile) [inline src/main/java/ua/com/conductor/dao/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't insert movie entity " + movie, e);
```

File: src/main/java/ua/com/conductor/exception/DataProcessingException.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package ua.com.conductor.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String message) {
```
Comment (Sofasmile) [inline src/main/java/ua/com/conductor/exception/DataProcessingException.java]:
```suggestion
    public DataProcessingException(String message, Throwable e) {
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**